src/software/physics/euclidean_to_wheel.h

```diff
@@ -54,13 +57,47 @@ class EuclideanToWheel
      * @param wheel_velocity The wheel velocity.
      * @return The equivalent Euclidean velocity.
      */
-    EuclideanSpace_t getEuclideanVelocity(const WheelSpace_t &wheel_velocity) const;
+    EuclideanSpace_t getEuclideanVelocity(const WheelSpace_t& wheel_velocity) const;
+
+
+    /**
+     * Ramp the velocity over the given timestep and set the target velocity on the motor.
+     *
+     * NOTE: This function has no state.
+     * Also NOTE: This function handles all electrical rpm to meters/second conversion.
+     *
+     * @param target_euclidean_velocity The target Euclidean velocity in m/s
+     * @param current_wheel_velocity The current 4-wheel velocity m/s
+     * @param time_to_ramp The time allocated for acceleration in seconds
+     *
+     */
+    WheelSpace_t rampWheelVelocity(const WheelSpace_t& current_wheel_velocity,
+                                   const EuclideanSpace_t& target_euclidean_velocity,
+                                   const double& time_to_ramp);
+
+    /**
+     * Convert the given Vector-Angle pair and Primitive to WheelSpace_t and
+     * EuclideanSpace_t to pass into rampWheelVelocity
+     *
+     * Modify given primitive with values from resulting WheelSpace_t and return pointer
+     * to it
+     *
+     * @param target_velocity_primitive The target DirectControl Primitive
+     * @param current_primitive The current velocity and Angular velocity
+     * @param time_to_ramp The time allocated for acceleration in seconds
+     *
+     */
+    std::unique_ptr<TbotsProto::DirectControlPrimitive> rampWheelVelocity(
+        const std::pair<Vector, AngularVelocity> current_local_velocity,
+        TbotsProto::DirectControlPrimitive& target_velocity_primitive,
+        const double& time_to_ramp);
 
    private:
     /**
      * The radius of the robot in meters.
      */
     const double robot_radius_m_{};
+    const RobotConstants_t robot_constants;
```

Comment:
nit: Just to stay consistent
```suggestion
    const RobotConstants_t robot_constants_;
```

Reply:
done